### PR TITLE
Lobby Hive Leader Viewing

### PR DESCRIPTION
**Pull request by Geevies**
## About The Pull Request

image

## Why It's Good For The Game

Xeno players have been asking for this for a while, especially with how much power a Queen has over a round.

## Changelog

:cl:
add: You can now see who's playing the Queen / her hive leaders from the lobby.
/:cl:

### CODE REVIEW

**Comment by c4xmaniac** [comment]
XX prime queen nerf

**Comment by stanalbatross** [comment]
All new player facing UIs must be TGUI

File: code/modules/mob/living/carbon/xenomorph/Xenomorph.dm [hunk 1]
```diff
@@ -531,8 +532,8 @@
 	if(client)
 		name_client_prefix = "[(client.xeno_prefix||client.xeno_postfix) ? client.xeno_prefix : "XX"]-"
 		name_client_postfix = client.xeno_postfix ? ("-"+client.xeno_postfix) : ""
-
 		age_xeno()
+	full_designation = "[name_client_prefix][nicknumber][name_client_postfix]"
```
Comment (stanalbatross) [inline code/modules/mob/living/carbon/xenomorph/Xenomorph.dm:536]: put maturity in maybe?

**Review by stanalbatross — COMMENTED** [review]
I can write a nicer version of the tgui probably

**Comment by morrowwolf** [comment]
Drafting til TGUI

File: code/modules/mob/living/carbon/xenomorph/Xenomorph.dm (hunk 1 above)
Comment (Geevies) [reply code/modules/mob/living/carbon/xenomorph/Xenomorph.dm:536]: nah, i think it's fine as-is

File: code/modules/mob/new_player/new_player.dm [hunk 2]
```diff
@@ -402,6 +406,39 @@
...
+/datum/hive_leaders/tgui_interact(mob/user, datum/tgui/ui)
+	ui = SStgui.try_update_ui(user, src, ui)
+	if(!ui)
+		ui = new(user, src, "HiveLeaders", "Hive Leaders")
+		ui.open()
+		ui.set_autoupdate(FALSE)
+
+// Player panel
+/datum/hive_leaders/ui_data(mob/user)
+	. = list()
```
Comment (stanalbatross) [inline code/modules/mob/new_player/new_player.dm:426]: data = list()

hijacking . makes it harder to read

**Review by stanalbatross — CHANGES_REQUESTED** [review]

File: code/modules/mob/new_player/new_player.dm [hunk 3]
```diff
@@ -52,6 +52,7 @@
 
 	else
 		output += "<a href='byond://?src=\ref[src];lobby_choice=manifest'>View the Crew Manifest</A><br><br>"
+		output += "<a href='byond://?src=\ref[src];lobby_choice=hiveleaders'>View Hive Leaders</A><br><br>"
```
[the remark this answers is not in the document]
Comment (Geevies) [reply code/modules/mob/new_player/new_player.dm:55]: that'd mean we have to update everyone in the lobby's thing to consider whether mainhive exists, so nah

[the remark this answers is not in the document]
Comment (stanalbatross) [reply code/modules/mob/new_player/new_player.dm:55]: but that already happens with join the hunt and pred rounds 🤔

**Review by stanalbatross — APPROVED** [review]

**Pull request merged**